Solutions/Marain.Tenancy.Specs/Integration/Features/TenancyClientWithCachingEnabled.feature

```diff
@@ -0,0 +1,88 @@
+﻿@perFeatureContainer
+@withTenancyClient
+@useTenancyFunction
+
+# Note that this set of specs does not include "create, update and retrieve" scenarios that are present in the
+# "TenancyClient with caching disabled" specs. This is because we don't recommend using caching in "management
+# plane" scenarios where tenants are being created, updated and so on.
+Feature: TenancyClient with caching enabled
+	In order to use Marain Tenant services
+	As a developer
+	I want to be able to access the standard ITenantProvider via the client API.
+
+Scenario: Get a tenant that does not exist
+	When I get a tenant with id "NotFound"
+	Then it should throw a TenantNotFoundException
+
+Scenario: Getting a tenant that has already been retrieved uses the cache
+	Given I create a child tenant called "ChildTenant1" for the root tenant
+	And I get the tenant id of the tenant called "ChildTenant1" and call it "ChildTenantId"
+	When I use the client to get the tenant with the id called "ChildTenantId" and call the response "Response1"
+	Then the tenant response called "Response1" was retrieved from the cache
+
+Scenario: Get a tenant with an etag retrieved from a created tenant
+	Given I create a child tenant called "ChildTenant1" for the root tenant
+	And I get the tenant id of the tenant called "ChildTenant1" and call it "ChildTenantId"
+	And I get the ETag of the tenant called "ChildTenant1" and call it "ChildTenantETag"
+	When I get the tenant with the id called "ChildTenantId" and the ETag called "ChildTenantETag"
+	Then it should throw a TenantNotModifiedException
+
+Scenario: Get a tenant with an etag retrieved from a tenant got from the repo
+	Given I create a child tenant called "ChildTenant1" for the root tenant
+	And I get the tenant id of the tenant called "ChildTenant1" and call it "ChildTenantId"
+	And I get the tenant with the id called "ChildTenantId" and call it "Result"
+	And I get the ETag of the tenant called "Result" and call it "ResultETag"
+	When I get the tenant with the id called "ChildTenantId" and the ETag called "ResultETag"
+	Then it should throw a TenantNotModifiedException
```

Comment:
Isn't "tenant not modified" the normal case? Throwing an exception seems odd here.

That said, I'm guessing this behaviour wasn't actually a change introduced in this PR?

Reply:
Nope, that's been there for ages.